modules/hcp-ec2-client/main.tf

```
@@ -44,18 +48,49 @@ resource "aws_security_group_rule" "allow_http_inbound" {
   protocol    = "tcp"
   cidr_blocks = var.allowed_http_cidr_blocks
 
-  security_group_id = var.security_group_id
+  security_group_id = aws_security_group.hcp_consul_ec2.id
+}
+
+#The bellow is to setup the instance profile and iam role to enable SSM
+resource "aws_iam_instance_profile" "hcp-ec2-iam-profile" {
```

nit: use all underscore for resource names, and no need to respecify `iam_profile`: https://www.terraform-best-practices.com/naming

```suggestion
resource "aws_iam_instance_profile" "hcp_ec2" {
```